### ISSUE TITLE
GEO Sql Failures due to serialization problems

### ISSUE THREAD

**Issue by benwtrent** [elastic/elasticsearch#44517]
Since the backport of #44490 there have been two build failures:
* https://scans.gradle.com/s/wg2a224t2lerw
* https://scans.gradle.com/s/vnnomcbkgyedw/

I could not reproduce it locally. Unsure how the serialization is failing randomly. Possibly an earlier running test is causing the stream to be corrupted?

Relevant reproduce lines:
```
./gradlew :x-pack:plugin:sql:qa:multi-node:integTestRunner --tests "org.elasticsearch.xpack.sql.qa.multi_node.GeoJdbcSqlSpecIT" \
  -Dtests.method="test {ogc.testSelectPondsXY}" \
  -Dtests.seed=4C0D5E2C7B7393A0 \
  -Dtests.security.manager=true \
  -Dtests.locale=ko-KR \
  -Dtests.timezone=America/Cayman \
  -Dcompiler.java=12 \
  -Druntime.java=8
```
```
./gradlew :x-pack:plugin:sql:qa:multi-node:integTestRunner --tests "org.elasticsearch.xpack.sql.qa.multi_node.GeoJdbcCsvSpecIT" \
  -Dtests.method="test {geosql.testSelectFilterByRegionPoint}" \
  -Dtests.seed=4C0D5E2C7B7393A0 \
  -Dtests.security.manager=true \
  -Dtests.locale=et-EE \
  -Dtests.timezone=America/Ojinaga \
  -Dcompiler.java=12 \
  -Druntime.java=8
```
```
./gradlew :x-pack:plugin:sql:qa:multi-node:integTestRunner --tests "org.elasticsearch.xpack.sql.qa.multi_node.GeoJdbcSqlSpecIT" \
  -Dtests.method="test {ogc.testSelectMapNeatLinesXY}" \
  -Dtests.seed=E2DC7B3731350E3C \
  -Dtests.security.manager=true \
  -Dtests.locale=es-CU \
  -Dtests.timezone=Europe/Warsaw \
  -Dcompiler.java=12 \
  -Druntime.java=8
```
```
./gradlew :x-pack:plugin:sql:qa:multi-node:integTestRunner --tests "org.elasticsearch.xpack.sql.qa.multi_node.GeoJdbcCsvSpecIT" \
  -Dtests.method="test {geosql.testGroupByEastWest}" \
  -Dtests.seed=E2DC7B3731350E3C \
  -Dtests.security.manager=true \
  -Dtests.locale=mk-MK \
  -Dtests.timezone=Asia/Vientiane \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

All of these tests failed the build due to a serialization problem with a `indices:admin/freeze` request.

```

»    ↓ errors and warnings from /var/lib/jenkins/workspace/elastic+elasticsearch+7.x+multijob+fast+part2/x-pack/plugin/sql/qa/multi-node/build/testclusters/integTest-1/logs/es.stdout.log ↓
--
» WARN ][o.e.d.FileBasedSeedHostsProvider] [integTest-1] expected, but did not find, a dynamic hosts list at [/var/lib/jenkins/workspace/elastic+elasticsearch+7.x+multijob+fast+part2/x-pack/plugin/sql/qa/multi-node/build/testclusters/integTest-1/config/unicast_hosts.txt]
» WARN ][r.suppressed             ] [integTest-1] path: /frozen_emp/_freeze, params: {index=frozen_emp}
»  org.elasticsearch.transport.RemoteTransportException: [integTest-0][127.0.0.1:39761][indices:admin/freeze]
»  Caused by: java.lang.IllegalStateException: Message not fully read (request) for requestId [1214], action [indices:admin/freeze], available [20]; resetting
»  	at org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:186) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:121) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:105) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
»  	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:661) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
»  	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:62) [transport-netty4-client-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) [netty-codec-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297) [netty-codec-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:241) [netty-handler-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:682) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:582) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:536) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [netty-transport-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:906) [netty-common-4.1.36.Final.jar:4.1.36.Final]
»  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.36.Final.jar:4.1.36.Final]
»  	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

**Comment by benwtrent** [comment]
Sorry for pinging both teams. I was not sure who to alert :/

**Comment by benwtrent** [comment]
Another failed build: https://scans.gradle.com/s/gq4tpfp5zm6oe

**Comment by ywelsch** [comment]
This should be fixed by #44454. I'm surprised this wasn't caught by the CI run here: https://github.com/elastic/elasticsearch/pull/44490
